base_layer/wallet/src/output_manager_service/handle.rs

```diff
@@ -54,8 +55,8 @@ pub enum OutputManagerRequest {
     GetCoinbaseTransaction((u64, MicroTari, MicroTari, u64)),
     ConfirmPendingTransaction(u64),
     ConfirmTransaction((u64, Vec<TransactionInput>, Vec<TransactionOutput>)),
-    PrepareToSendTransaction((MicroTari, MicroTari, Option<u64>, String)),
-    CreatePayToSelfTransaction((MicroTari, MicroTari, Option<u64>, String)),
+    PrepareToSendTransaction((MicroTari, MicroTari, Option<u64>, String, TariScript)),
+    CreateOneSidedTransaction((MicroTari, MicroTari, Option<u64>, String, Option<RistrettoPublicKey>)),
```

Comment:
So as discussed I think this special case shouldn't be handled in the OMS and you just make the `PrepareToSendTransaction` to accept a script as you have above